cpp/oneapi/dal/algo/pca/backend/cpu/infer_kernel_cov.cpp

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
```

Comment:
I'd suggest to not separate `cov` and `svd` methods into two distinct files. The algorithm for both methods are the same, so having `infer_kernel.cpp`  will be enough. 

In `infer_kernel.hpp`,  remove the `Method` template parameter at all:
```c++
template <typename Float, typename Task>
struct infer_kernel_gpu {
    infer_result<Task> operator()(const dal::backend::context_gpu& ctx,
                                  const descriptor_base<Task>& params,
                                  const infer_input<Task>& input) const;
};
```

So, call to the kernels in the dispatcher would be the following:
```c++
// No method passed to kernel from dispatcher
        using kernel_dispatcher_t =
            dal::backend::kernel_dispatcher<backend::infer_kernel_cpu<Float, Task>>;
```